aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java

```diff
@@ -68,6 +69,34 @@
     this.key = path;
   }
 
+  /**
+   * Creates a new S3URI in the form of scheme://bucket/key?query#fragment
+   * <p>
+   * The URI supports any valid URI schemes to be backwards compatible with s3a and s3n,
+   * and also allows users to use S3FileIO with other S3-compatible object storage services like GCS.
+   *
+   * @param location fully qualified URI
+   * @param bucketToAccessPointMapping contains mapping of bucket to access point
+   */
+  S3URI(String location, Map<String, String> bucketToAccessPointMapping) {
```

Comment:
I think the original constructor can just call `S3URI(location, ImmutableMap.of())`